set.mm

```diff
@@ -635310,6 +635400,361 @@ deduction version (Contributed by metakunt, 22-May-2024.) $)
+       metakunt, 28-May-2024.) $)
+    metakunt24 $p  |- ( ph
+           -> ( ( ( ( 1 ... ( I - 1 ) )
+                       u. ( I ... ( M - 1 ) ) )
+                   i^i { M } )
+                 = (/)
+              /\ ( 1 ... M )
+                 = ( ( ( 1 ... ( I - 1 ) ) u. ( I ... ( M - 1 ) ) )
+                   u. { M } )
+              /\ ( 1 ... M )
+                 = ( ( ( ( ( M - I ) + 1 ) ... ( M - 1 ) )
+                      u. ( 1 ... ( M - I ) ) )
+                              u. { M } ) ) ) $=
+      ( c1 cmin co cfz cun cin c0 wceq caddc eqtrd wcel adantr eqcomd cc0 indir
+      csn a1i w3a metakunt18 simpld simp2d simp3d uneq12d unidm 1zzd nnzd nnred
+      nnge1d leidd elfzd fzsplitnd oveq1 oveq2d uneq1d adantl clt ltm1d zsubcld
+      wa wbr cz wb fzn syl2anc mpbid uneq2d un0 wn cle wne eqeq1 mpbii necon3bi
+      eqid leltned mpbird fzsplitnr pm2.61dan fzsn syl 3eqtrd uncom fz10 uneq1i
+      zltlem1 eqeq1i imbi2i mpbi oveq2 recnd subidd oveq1d 1e0p1 syl6eqr lesubd
+      wi cr 1red lesub2dd fzsplit 3jca ) AGBGHIZJIZBCGHIZJIZKZCUBZLZMNGCJIZXLXM
+      KZNXOCBHIZGOIZXJJIZGXQJIZKZXMKZNAXNXIXMLZXKXMLZKZMXNYENAXIXKXMUAUCAYEMMKZ
+      MAYCMYDMAXIXKLMNZYCMNZYDMNZAYGYHYIUDXSXTLMNXSXMLMNXTXMLMNUDABCDEFUEUFZUGA
+      YGYHYIYJUHUIYFMNAMUJUCPPAXOGXJJIZCCJIZKZXLYLKZXPACGCACGCAUKZACDULZYPACDUN
+      ACACDUMZUOUPUQZABCNZYMYNNAYSVEZYNYMYTYNYKCXJJIZKZYLKZYMYSYNUUCNAYSXLUUBYL
+      YSXIYKXKUUAYSXHXJGJBCGHURUSBCXJJURUIUTVAYTUUBYKYLYTUUBYKMKZYKYTUUAMYKAUUA
+      MNZYSAXJCVBVFZUUEACYQVCACVGQZXJVGQUUFUUEVHYPACGYPYOVDCXJVIVJVKRVLUUDYKNZY
+      TYKVMUCZPUTPSAYSVNZVEZYKXLYLUUKBGXJAGVGQUUJYORZUUKCGAUUGUUJYPRZUULVDZABVG
+      QZUUJABEULZRZAGBVOVFUUJABEUNRZUUKBCVBVFZBXJVOVFZUUKUUSCBVPZUUJUVAAYSCBCBN
+      CCNYSCVTCBCVQVRVSVAAUUSUVAVHUUJABCABEUMZYQFWARWBAUUSUUTVHZUUJAUUOUUGUVCUU
+      PYPBCWKVJRVKZWCUTWDAYLXMXLAUUGYLXMNYPCWEWFZVLWGAYBXOAYBXTXSKZXMKZXOAYAUVF
+      XMYAUVFNAXSXTWHUCUTAUVGUVFYLKZXOAUVHUVGAYLXMUVFUVEVLSAXOUVHAXOYMUVHYRAYKU
+      VFYLAYSYKUVFNZYTYKGTJIZYKKZUVFYTUVKYKYTUVKMYKKZYKAUVKUVLNZYSUVMAUVJMYKWIW
+      JUCRYTUUHXBYTUVLYKNZXBUUIUUHUVNYTUUDUVLYKYKMWHWLWMWNPSYTUVFUVKYTXTUVJXSYK
+      YTXQTGJYTXQCCHIZTYSXQUVONABCCHWOZVAAUVOTNYSACACYQWPWQRZPUSYTXRGXJJYTXRTGO
+      IZGYTXRUVOGOIZUVRYSXRUVSNAYSXQUVOGOUVPWRVAYTUVOTGOUVQWRPWSWTWRUISPUUKXQYK
+      QUVIUUKXQGXJUULUUNUUKCBUUMUUQVDUUKBCGABXCQUUJUVBRZACXCQUUJYQRZUUKXDZUVDXA
+      UUKGBCUWBUVTUWAUURXEUPXQGXJXFWFWDUTPSPPSXG $.
+  $}
+
+  ${
+    $d B y $.  $d C y $.  $d D y $.  $d I x $.  $d M x y $.  $d ph x y $.
+    metakunt25.1 $e |- ( ph -> M e. NN ) $.
+    metakunt25.2 $e |- ( ph -> I e. NN ) $.
+    metakunt25.3 $e |- ( ph -> I <_ M ) $.
+    metakunt25.4 $e |- B = ( x e. ( 1 ... M ) |-> if ( x = M , M ,
+     if ( x < I , ( x + ( M - I ) ) , ( x + ( 1 - I ) ) ) ) ) $.
+    metakunt25.5 $e |- C = ( x e. ( 1 ... ( I - 1 ) ) |->
+     ( x + ( M - I ) ) ) $.
+    metakunt25.6 $e |- D = ( x e. ( I ... ( M - 1 ) ) |->
+     ( x + ( 1 - I ) ) ) $.
```

Comment:
C and D are extra hypotheses (they don't appear in the final statement). They may _optionally_ be removed with eqid in the proof itself if you wish.

Example: https://us.metamath.org/mpeuni/mhpdeg.html
Pro is less hypotheses for future theorems, con is longer proof (at minimum by syntax encoding and referencing eqid; in a compressed proof, a hypothesis is one letter unless there are an incredible amount of variables and hypotheses)

Reply:
I have implemented that suggestion. In fact it was always my goal to not have them in the final step. I must have forgotten it due to the proof being such a colossal pain that I have forgotten I am at the last step.